src/typing/flow_js.ml

```diff
@@ -3644,6 +3644,15 @@ let rec __flow cx ((l: Type.t), (u: Type.use_t)) trace =
       in
       rec_flow cx trace (o, u)
 
+    | (NumT (reason_x, Literal (n, _)), ElemT(reason_op, (ObjT _ as o), t, rw)) ->
+      let x = string_of_float_trunc n in
```

Comment:
Hi Vladimir - it looks like the runtime behavior of e.g. `obj[n]` is to call `toString` on `n` and use the result as a property name, so `string_of_float_trunc` doesn't seem right here - instead we would need some `js_string_of_float` which replicated the specified behavior of JS's `(number).toString`.

(Aside: with that in hand, it should be pretty straightforward to support numeric property names in object literals as well!)